g4f/Provider/DDG.py

```diff
@@ -74,32 +76,38 @@ async def create_async_generator(
         **kwargs
     ) -> AsyncResult:
         async with aiohttp.ClientSession(headers=cls.headers, connector=get_connector(connector, proxy)) as session:
-            vqd_4 = None
-            if conversation is not None and len(messages) > 1:
+            if len(messages) > 1:
+                if conversation is None:
+                    raise ValueError(f"More than one message requires using conversation.")
+
                 vqd_4 = conversation.vqd_4
-                messages = [*conversation.messages, messages[-2], messages[-1]]
             else:
                 for _ in range(3):  # Try up to 3 times to get a valid VQD
                     vqd_4 = await cls.get_vqd(session)
                     if vqd_4:
                         break
                     await asyncio.sleep(1)  # Wait a bit before retrying
-                
+
                 if not vqd_4:
                     raise Exception("Failed to obtain a valid VQD token")
-                
-                messages = [messages[-1]]  # Only use the last message for new conversations
-            
+
             payload = {
                 'model': cls.get_model(model),
                 'messages': [{'role': m['role'], 'content': m['content']} for m in messages]
             }
-            
+
             async with session.post(cls.chat_url, json=payload, headers={"x-vqd-4": vqd_4}) as response:
+                if response.status == 400:
+                    if "ERR_INVALID_VQD" in await response.text():
+                        raise ResponseError(
+                            f"Status {response.status}: Conversation does not exactly match the messages or its id (vqd-4) is invalid"
+                        )
                 await raise_for_status(response)
                 if return_conversation:
-                    yield Conversation(vqd_4, messages)
-                
+                    vqd_4_new = response.headers.get("x-vqd-4")
+                    if vqd_4_new is not None:
+                        yield Conversation(vqd_4_new, messages)
```

Comment:
Should I raise an exception if we could not get the conversation id, but the user requested it? Currently it's not a silent error, the user can detect an absence of the Converation chunk.